### PR TITLE
[fix] categories.html: can't select social media category using search on category select

### PR DESCRIPTION
**Pull request by Bnyro**
## What does this PR do?
* fixes a regression from #2740 that we've overseen

## How to test this PR locally?
* make run
* search anything and click on the "social media" category

### CODE REVIEW

File: searx/templates/simple/categories.html [hunk 1]
```diff
@@ -26,7 +26,7 @@
             {%- if display_tooltip %}<div class="help">{{ _('Click on the magnifier to perform search') }}</div>{% endif -%}
         {%- else -%}
             {%- for category in categories_as_tabs -%}{{- '\n' -}}
-                <button type="submit" name="category_{{ category|replace(' ', '_') }}" class="category category_button {% if category in selected_categories %}selected{% endif %}">
+                <button type="submit" name="category_{{ category }}" class="category category_button {% if category in selected_categories %}selected{% endif %}">
```
Comment (dalf) [inline searx/templates/simple/categories.html:29]: The line 19 has exactly the same `|replace(' ', '_')` but it is left untouched, is it intended?

Comment (Bnyro) [reply searx/templates/simple/categories.html:29]: Yes, I don't really understand why but if you disable search on category select, it still works fine when selecting social media.

Comment (return42) [reply searx/templates/simple/categories.html:29]: > The line 19 has exactly the same `|replace(' ', '_')` but it is left untouched, is it intended?

In line 19 the *replace* is done on the `id`, but not on the `name` attribute:

`id="checkbox_{{ category|replace(' ', '_') }}" name="category_{{ category }}`

The `name` attribute is used for the preferences .. and if I'm not wrong, in the preferences code we use the category *name* without replacing the white-spaces:

https://github.com/searxng/searxng/blob/e37d775fa2da6de818e0dc73842bbb0bddc615ef/searx/preferences.py#L295-L299

For instance `social media` from here:

https://github.com/searxng/searxng/blob/e37d775fa2da6de818e0dc73842bbb0bddc615ef/searx/settings.yml#L285-L295

**Review by return42 — APPROVED** [review]
can confirm, this PR fixes the issue :+1:

**Pull request merged**